package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
With all the thinking, I might have not been clear with the resulting desire:
```js
/**
 * Gets a _foo_ given a bar.
 * @param {object} bar A bar to get a foo from.
 * @return {object} A foo.
 * @throws {Error} Invalid argument. When `bar` is closed.
 */
function foo(bar) {
  if(bar == null) throw error.argRequired("bar");
  if(DEBUG) {
    if(bar.closed) throw error.argInvalid("bar", "Bar is closed. Cannot get foo.");
  }

  // ...
}
```

So:
1. Implement `argRequired` errors
2. Implement `argInvalidType` errors in some cases; preferably wrapping in DEBUG block
3. Don't document `argRequired` or `argInvalidType` errors (they're implied in the contract)
4. Document and implement `argInvalid`, `operInvalid` errors (without DEBUG block!)

What about the use of `@throws`?